lib/permissions.spec.ts

```diff
@@ -0,0 +1,352 @@
+import { residentFactory } from '../factories/residents';
+import { userFactory } from '../factories/users';
+
+import { canUserEditPerson, canManageCases } from './permissions';
+
+describe('permissions', () => {
+  describe('#canUserEditPerson()', () => {
+    it('should return false when the user has no permissions', () => {
+      expect(
+        canUserEditPerson(
+          userFactory.build({
+            hasAdminPermissions: false,
+            hasAdultPermissions: false,
+            hasChildrenPermissions: false,
+            hasUnrestrictedPermissions: false,
+            hasDevPermissions: false,
+            hasAllocationsPermissions: false,
+          }),
+          residentFactory.build({
+            restricted: 'N',
+            contextFlag: 'C',
+          })
+        )
+      ).toEqual(false);
+    });
+
+    it('should return false when the resident is an adult', () => {
```

Comment:
Would this mean that users would not be able to edit an adult person only if they are in adults? 
Or would this be regardless of permissions too? 🤔

Reply:
Adults currently can't be edited at all, no matter what your permission is, so that's what this test is confirming.

Aside: I gather ASC have actually just signed off adult editing, so we will probably be changing this shortly anyway, https://github.com/LBHackney-IT/lbh-social-care-frontend/pull/365